scalafix-core/shared/src/main/scala/scalafix/rule/Rule.scala

```diff
@@ -42,7 +42,8 @@ import metaconfig.Configured
   *   Example good name: NoVars, ExplicitUnit.
   *   Example bad name: no-vars, noVars, FixVars.
   */
-abstract class Rule(ruleName: RuleName) { self =>
+abstract class Rule(ruleName: RuleName, ruleDescription: Option[String]) {
```

Comment:
This line here is causing the breaking changes, we cannot change the constructor for `Rule`. scalafix.Rule must take extra care for source compatibility since we compile rules from source from external repos like https://github.com/typelevel/cats/blob/66528459d8cec954dc01dcaa799b5cfe4db8ff6a/scalafix/rules/src/main/scala/fix/Cats_v1_0_0.scala

Reply:
Sorry, I am not proposing this as the final solution. I meant just to apply the easy fixes, and then get incrementally back to the single constructor. If you prefer, I will avoid to publish my incremental improvements.

Reply:
My apologies! I thought this was your final commit, feel free to push intermediary commits. Just ping me when you think this is ready for another round of review 😄

Reply:
Great! I will write a comment at the bottom as soon as I have something I think works.
(By the way, I have still to do some work on the completion bits to solve the original issue :P)